Sources/PrimerSDK/Classes/User Interface/Text Fields/CardComponentsManager.swift

```diff
@@ -332,6 +338,109 @@ public class CardComponentsManager: NSObject, CardComponentsManagerProtocol {
+            
+            guard let paymentMethod = paymentMethod else {
+                let err = PrimerError.invalidValue(key: "paymentMethod")
+                Primer.shared.delegate?.onResumeError?(err)
+                handle(error: err)
+                return
+            }
+           
+            if decodedClientToken.intent == RequiredActionName.threeDSAuthentication.rawValue {
+                #if canImport(Primer3DS)
+                let threeDSService = ThreeDSService()
+                threeDSService.perform3DS(
+                    paymentMethodToken: paymentMethod,
+                    protocolVersion: state.decodedClientToken?.env == "PRODUCTION" ? .v1 : .v2,
+                    beginAuthExtraData: nil,
+                    sdkDismissed: nil)
+                { result in
+                    switch result {
+                    case .success(let paymentMethodToken):
+                        guard let threeDSPostAuthResponse = paymentMethodToken.1,
+                              let resumeToken = threeDSPostAuthResponse.resumeToken else {
+                            let err = PrimerError.threeDSFailed
+                            Primer.shared.delegate?.onResumeError?(err)
+                            self.handle(error: err)
+                            return
+                        }
+                       
+                        Primer.shared.delegate?.onResumeSuccess?(resumeToken, resumeHandler: self)
+                       
+                    case .failure(let err):
+                        log(logLevel: .error, message: "Failed to perform 3DS with error \(err as NSError)")
+                        let err = PrimerError.threeDSFailed
+                        Primer.shared.delegate?.onResumeError?(err)
+                        self.handle(error: err)
+                    }
+                }
+                #else
+                let error = PrimerError.threeDSFailed
+                Primer.shared.delegate?.onResumeError?(error)
+                #endif
+               
+            } else {
+                let err = PrimerError.invalidValue(key: "resumeToken")
+                Primer.shared.delegate?.onResumeError?(err)
+                handle(error: err)
+            }
+           
+        } catch {
+            Primer.shared.delegate?.onResumeError?(error)
+            handle(error: error)
+        }
+    }
+    
+    public func handleSuccess() {
+        DispatchQueue.main.async {
+            self.setIsLoading(false)
+            
+            let settings: PrimerSettingsProtocol = DependencyContainer.resolve()
+
+            if settings.hasDisabledSuccessScreen {
+                Primer.shared.dismiss()
+            } else {
+                let svc = SuccessViewController()
+                svc.view.translatesAutoresizingMaskIntoConstraints = false
+                svc.view.heightAnchor.constraint(equalToConstant: 300).isActive = true
+                Primer.shared.primerRootVC?.show(viewController: svc)
```

Comment:
[Q]
If I'm not mistaken, the `CardComponentsManager` is the checkout components for mobile?
If we ship the `CardComponentsManager` separately for merchants, that means that the manager automatically displays a `SuccessViewController` upon success? 
Don't think that's a good solution either, it's tightly coupling this standalone manager to the UI logic of the Universla Checkout 🤔